src/core/create_compilers/RollupCompiler.ts

```diff
@@ -69,13 +88,27 @@ export default class RollupCompiler {
 	}
 
 	async get_config(mod: any) {
+
+		const onwarn = mod.onwarn || ((warning: any, handler: (warning: any) => void) => {
+			handler(warning);
+		});
+
+		mod.onwarn = (warning: any) => {
+			onwarn(warning, (warn: any) => {
+				this.warnings.push(warn);
+			});
+		};
+
 		let entry_point: string | undefined;
 
 		const that = this;
-		const sourcemap = mod.output.sourcemap;
 
-		// TODO this is hacky, and doesn't need to apply to all three compilers
-		(mod.plugins || (mod.plugins = [])).push({
+		// TODO this is hacky. refactor out into an external rollup plugin
+		(mod.plugins || (mod.plugins = [])).push(css_chunks({ injectImports: true }));
+		if (!JSON.stringify(mod.input).includes('client.')) {
```

Comment:
My attention was drawn here because it was the line involved in the fix for Windows - and the new version of it makes me even more uncomfortable than the old version.

Why are we looking at `JSON.stringify(mod.input)`? Are there multiple fields within `mod.input` where this filename could occur? And is there a more precise regex we could use here, like `/\/client\/client\.\w+\.js$/` or something?

Reply:
yeah, this whole section is a bit icky. it wasn't the "TODO this is hacky" comment that brought your attention here? :smile: 

ok, I just updated it to be a bit better

I hope that we can refactor this out into an external plugin (either make it part of `rollup-plugin-css-chunks` or a new one) very shortly which would make this go away